src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -151,7 +151,16 @@ static optionalt<std::string> extract_generic_interface_reference(
     start =
       find_closing_semi_colon_for_reference_type(signature.value(), start) + 1;
 
-    start = signature.value().find("L" + interface_name + "<", start);
+    // if the interface name includes package name, convert dots to slashes
+    std::string interface_name_slash_to_dot = interface_name;
```

Comment:
When doing this kind of string manipulation, it is probably worth also checking that inner classes work (since the whole `$`/`.` fiasco) 

For example (though I guess with some generics in there as well). 
```
class Outer {
  public interface InnerInterface { }
}
class TestClass implements Outer.InnerInterface {}
```

Reply:
I tried the example, this is not causing a problem since both in class signatures and in our representation we use `$` as separator for inner classes.